src/components/panel/panel.js

```diff
@@ -464,19 +484,34 @@ function MdPanelRef(config, $injector) {
  * is opened and animations finish.
  */
 MdPanelRef.prototype.open = function() {
-  if (!this.isOpen) {
-    this.isOpen = true;
+  if (this.isOpen) {
+    // Ignore open. We don't want chaining to happen here.
+    return this._$q.reject('Panel is already opened.');
+  }
+
+  if (this._openPromise) {
+    // If the panel is already open, re-use the (already resolved) promise from
+    // when it was opened.
+    return this._openPromise;
+  }
+
+  // TODO(ErinCoughlan) - Cancel any in-progress `closing`.
 
-    var self = this;
-    this._openPromise = this._$q(function(resolve, reject) {
-      self._openReject = reject;
+  this.isOpen = true;
+  var self = this;
+  this._openPromise = this._$q(function(resolve, reject) {
+    try {
```

Comment:
What's the reasoning behind the try-catch here? This will eat any errors that are thrown while opening so that they're not logged to the console.